CON-229: modularize avail node content (#1218)
maticnetwork/matic-docs: docs/avail/node/avail-validator.md

Validator nodes are responsbile for maintaining and securing the network by staking tokens with real
value. Validators need to understand how to manage their node, its associated hardware & configuration,
and be wary that they are subject to being slashed due to actions like being offline or equivocation.

When in doubt, reach out to the Validator Engagement team.

:::

## Docker Setup

The easiest way to deploy your own Avail validator node is using Docker.

### Run the latest version of the Docker container

Use the default parameters and expose the P2P port with `-p 30333` by running:

```shell
docker run -p 30333 --name my_val 0xpolygon/avail:latest
```

Any extra parameter will be added to the `data-avail` binary as an argument.
If you want to use a specific node key and limit the maximum number of incoming connections
to `10`, you can use:

```shell
docker run -p 30333 --name my_val 0xpolygon/avail:latest --in-peers=10 --node-key 80027666cebec66464611eb0d5c36416213d83a9c689006a80efcf479826de7d
```

This image uses two volumes:
  - `/da/state` to store the database of the chain
  - `/da/keystore` to store the validator's private keys

> Most likelihood you want to bind these volumes to a specific points, like:

Reviewer: ```suggestion
Most likely you want to bind these volumes to a specific point, like:
```

```shell
docker run -p 30333 --name my_val -v /volumes/da/state:/da/state -v /volumes/da/keystore/:/da/keystore 0xpolygon/avail:latest
```

### Insert private keys

These private keys will be used by the validator to sign blocks and finalize the chain when it
acts as an active validator. They are stored into `/da/keystore` in plain text format, so you
should take extra care over that volume.

In order to insert these keys, we will open a shell inside the running container:

```shell
docker exec -it my_val bash
root@5f55e51e5a85:/da# /da/bin/data-avail key insert \
      --chain=/da/genesis/testnet.chain.spec.raw.json \
      --base-path=/da/state/ \
      --keystore-path=/da/keystore/ \
      --suri=0x7d98...cae6 \
      --key-type=babe \
      --scheme=Sr25519
```

> The **--suri** parameter is the private key as a mnemonic phrase (or secret phrase) where you can generate

Reviewer: ```suggestion
The `--suri` parameter is the private key as a mnemonic phrase (or secret phrase) where you can generate
```

one using the `subkey` tool in Substrate.

:::note Learn about subkey

> To learn about how to use subkey, visit the [Subkey Substrate documentation](https://docs.substrate.io/v3/tools/subkey/).

Reviewer: ```suggestion
To learn about how to use the subkey, visit the [<ins>Subkey Substrate documentation</ins>](https://docs.substrate.io/v3/tools/subkey/).
```

:::

This command should be **repeated for each pair of key type and scheme** shown in the following table:

| Key Type | Scheme    |
| -------- | --------- |
| babe     | Sr25519   |
| gran     | *Ed25519* |
| imon     | Sr25519   |
| audi     | Sr25519   |

## Bond AVL tokens

It is highly recommended that you set up a stash and controller account and have separate key (two separate accounts) for both.

:::info Stash and Controller Keys

- A controller key is used to control staking actions for your account

> - A stash key is used to control your funds. **It is recommended that the stash key be a cold wallet or offline and not be used for account related activities like submitting extrinsics.

Reviewer: ```suggestion
- A stash key is used to control your funds. It is recommended that the stash key be a cold wallet or offline and not be used for account-related activities like submitting extrinsics.
```

to learn more about stash and controller accounts and how to manage them.

:::

You will start by creating two accounts; ensure each account has enough funds to pay the fees for
making transactions.

:::tip Storing Funds

Keep most of your funds in the stash account since it is meant to be the custodian of
your staking funds, and have just enough funds in the controller account to pay for fees.

Make sure not to bond all your AVL balance since you will be unable to pay transaction fees from your bonded
balance.

:::

It is now time to set up your validator by doing the following:

 - Bond the AVL of the Stash account. These token will be put at stake for the security of the network and
   subject to slashing.
 - Select the Controller. This is the account that will decide when to start or stop validating.

First, go to the **Developer** tab in the [Avail Apps](https://devnet-avail.polygon.technology/)
navbar and click on **Extrinsics**.

* **Stash** account - Select your Stash account. In this example, we bond 1001 AVL tokens, where the

>   minimum bonding amount is 1000. Make sure that your Stash account contains at least this much.

Reviewer: ```suggestion
  minimum bonding amount is 1000. Make sure that your Stash account contains at least this many tokens.
```

<img src={useBaseUrl("img/avail/add_validator_bound_step.png")} width="100%" height="100%"/>

Create a transaction where your **stash** account bounds 1001 AVLs at least to your **controller** account,
as shown below.

<img src={useBaseUrl("img/avail/bond-avl-val.png")} width="100%" height="100%"/>

## Set Session Keys

Once your node is **fully synced**, you need to rotate and submit your session keys.

### Rotate you session keys

Run this command on the same machine (while the node is running with the default HTTP RPC port configured):

```shell
docker exec -it my_val bash
root@5f55e51e5a85:/da# curl \
      -H "Content-Type: application/json" \
      -d '{"id":1, "jsonrpc":"2.0", "method": "author_rotateKeys", "params":[]}' \
      http://localhost:9933
```

The output will have a hex-encoded "result" field. The result is the concatenation of the four public keys.
Save this result for a later step.

You can restart your node at this point.

### Submitting the `setKeys` transaction

You need to tell the chain your Session keys by signing and submitting an extrinsic. This is what associates
your validator with your Controller account.

> Navigate to the [Network > Staking](https://devnet-avail.polygon.technology/#/staking).

Reviewer: ```suggestion
Navigate to the [**Network &rarr; Staking**](https://devnet-avail.polygon.technology/#/staking).
```

> Here, you can perform various staking actions. Navigate to **Account actions** , and select **Set Session Key**

Reviewer: ```suggestion
Here, you can perform various staking actions. Navigate to **Account actions**, and select **Set Session Key**
```
